2012-01-20  Andrew Haley  <aph@redhat.com>

	* Makefile (ARCH): Add entry for armv7l.
	(CFLAGS): Enable debugging.
	($(LIBRARIES)): Pass CFLAGS to sub-make.
	* hsdis.c: Include string.h.
	(parse_caller_options): Fix strncmp bug.
	(native_arch_name): Add entry for ARM.

diff -u openjdk/hotspot/src/share/tools/hsdis/hsdis.c ../icedtea6/openjdk/hotspot/src/share/tools/hsdis/hsdis.c
--- openjdk/hotspot/src/share/tools/hsdis/hsdis.c	2011-11-14 17:07:33.000000000 -0500
+++ openjdk/hotspot/src/share/tools/hsdis/hsdis.c	2012-01-20 10:21:28.000000000 -0500
@@ -22,8 +22,6 @@
  *
  */
 
-#include "precompiled.hpp"
-
 /* hsdis.c -- dump a range of addresses as native instructions
    This implements the plugin protocol required by the
    HotSpot PrintAssembly option.
@@ -37,6 +35,8 @@
 #include <dis-asm.h>
 #include <inttypes.h>
 
+#include <string.h>
+
 #ifndef bool
 #define bool int
 #define true 1
@@ -358,7 +358,7 @@
       if (plen > mach_size)  plen = mach_size;
       strncpy(mach_option, p, plen);
       mach_option[plen] = '\0';
-    } else if (plen > 6 && strncmp(p, "hsdis-", 6)) {
+    } else if (plen > 6 && !strncmp(p, "hsdis-", 6)) {
       // do not pass these to the next level
     } else {
       /* just copy it; {i386,sparc}-dis.c might like to see it  */
@@ -420,6 +420,9 @@
 #ifdef LIBARCH_sparcv9
     res = "sparc:v9b";
 #endif
+#ifdef LIBARCH_arm
+    res = "arm";
+#endif
   if (res == NULL)
     res = "architecture not set in Makefile!";
   return res;
diff -u openjdk/hotspot/src/share/tools/hsdis/Makefile openjdk/hotspot/src/share/tools/hsdis/Makefile
--- openjdk/hotspot/src/share/tools/hsdis/Makefile	2011-11-14 17:07:33.000000000 -0500
+++ openjdk/hotspot/src/share/tools/hsdis/Makefile	2012-01-20 10:20:32.000000000 -0500
@@ -68,14 +68,18 @@
 CONFIGURE_ARGS= --host=$(MINGW) --target=$(MINGW)
 else
 CPU             = $(shell uname -m)
+ifneq		 ($(findstring arm,$(CPU)),)
+ARCH=arm
+else
 ARCH1=$(CPU:x86_64=amd64)
 ARCH=$(ARCH1:i686=i386)
 CFLAGS/i386	+= -m32
 CFLAGS/sparc	+= -m32
 CFLAGS/sparcv9	+= -m64
 CFLAGS/amd64	+= -m64
+endif
 CFLAGS		+= $(CFLAGS/$(ARCH))
-CFLAGS		+= -fPIC
+CFLAGS		+= -fPIC -g
 OS		= linux
 LIB_EXT		= .so
 CC 		= gcc
@@ -118,7 +122,7 @@
 BINUTILSDIR	= $(shell cd $(BINUTILS);pwd)
 endif
 
-CPPFLAGS	+= -I$(BINUTILSDIR)/include -I$(BINUTILS)/bfd -I$(TARGET_DIR)/bfd
+CPPFLAGS	+= -I$(BINUTILSDIR)/include -I$(BINUTILSDIR)/bfd -I$(TARGET_DIR)/bfd
 CPPFLAGS	+= -DLIBARCH_$(LIBARCH) -DLIBARCH=\"$(LIBARCH)\" -DLIB_EXT=\"$(LIB_EXT)\"
 
 TARGET_DIR	= build/$(OS)-$(JDKARCH)
@@ -145,7 +149,7 @@
 demo: $(TARGET) $(DEMO_TARGET)
 
 $(LIBRARIES): $(TARGET_DIR) $(TARGET_DIR)/Makefile
-	if [ ! -f $@ ]; then cd $(TARGET_DIR); make all-opcodes; fi
+	if [ ! -f $@ ]; then cd $(TARGET_DIR); make all-opcodes "CFLAGS=$(CFLAGS)"; fi
 
 $(TARGET_DIR)/Makefile:
 	(cd $(TARGET_DIR); CC=$(CC) CFLAGS="$(CFLAGS)" $(BINUTILSDIR)/configure --disable-nls $(CONFIGURE_ARGS))
